aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -91,7 +95,9 @@ pub fn allow_module_bundle_for_test() {
 }
 
 #[derive(Clone)]
-pub struct AptosVM(pub(crate) AptosVMImpl);
+pub struct AptosVM {
+    pub(crate) vm_impl: AptosVMImpl,
```

Comment:
Why do we do these indirections in the first place?

Reply:
I have no idea, and we should merge them

Reply:
I hated .0, but yes, this is complete and utter demonstration of ADHD.
I'll actually move it out, otherwise i'm adding so much noise to the already huge PR

Reply:
yes, also why do we need these indirections is a great point. I was just making a straightforward transformation, but hopefully we will change AptosVM implementation soon for the better